htdocs/web_portal/controllers/service_group/add_service_group.php

```diff
@@ -67,16 +69,30 @@ function submit($user) {
     }
 }
 
-function draw($user) {
+function draw($user)
+{
     try {
-        // If the user is registered they're allowed to add a service group
+        $hasRolesAssigned = false;
+        $isUserValid = false;
+        $hasAdminCredentials = $user->isAdmin();
+
         if (is_null($user)) {
             throw new \Exception("Unregistered users can't create service groups.");
         }
 
```

Comment:
Switching the order of these is probably safer, as there's no guarantee `$user` is not `NULL` when the isAdmin method gets called (though it does seem PHP will happily have things like `NULL->someMethod()` return `NULL`).

```suggestion
        if (is_null($user)) {
            throw new \Exception("Unregistered users can't create service groups.");
        }

        $hasRolesAssigned = false;
        $isUserValid = false;
        $hasAdminCredentials = $user->isAdmin();
```

Reply:
Updated the code.